packages/wavs/src/services.rs

```diff
@@ -91,30 +83,26 @@ impl Services {
         bounds_start: Bound<&ServiceId>,
         bounds_end: Bound<&ServiceId>,
     ) -> Result<Vec<Service>> {
-        let services = self
-            .db_storage
-            .with_table_read(&handles::SERVICES, |table| {
-                let mut services = BTreeMap::new();
-
-                for entry in table.iter() {
-                    let (key, value) = entry.pair();
-                    services.insert(key.clone(), value.clone());
-                }
+        let mut services = BTreeMap::new();
+
+        for entry in self.db_storage.services.iter() {
+            let (key, value) = entry.pair();
+            services.insert(key.clone(), value.clone());
+        }
 
-                let convert_bound = |bound: Bound<&ServiceId>| match bound {
-                    Bound::Unbounded => Bound::Unbounded,
-                    Bound::Included(id) => Bound::Included(id.clone()),
-                    Bound::Excluded(id) => Bound::Excluded(id.clone()),
-                };
+        let convert_bound = |bound: Bound<&ServiceId>| match bound {
+            Bound::Unbounded => Bound::Unbounded,
+            Bound::Included(id) => Bound::Included(id.clone()),
+            Bound::Excluded(id) => Bound::Excluded(id.clone()),
+        };
```

Comment:
`List` method first collects all services before applying range filter.
Since `ServiceId` implements `Ord` https://github.com/Lay3rLabs/WAVS/blob/c279a66724e6adf7a0c40eb56abad7a0523c8228/packages/types/src/id/hash.rs#L29 you should be able to filter it out during iteration, right?

Reply:
The main reason behind collecitng into a btreemap (from dashmap) is for making use of that ordering property. Otherwise bounds logic doesnt really make sense

Reply:
Yeah, I wanted to suggest a change to not extra allocate that BTreeMap but in the process I realized that it's still the most efficient way to go about it.